### PR TITLE
[WIP] Added handling of composite data types for comparison operator `$gt``$lt`, `$gte``$lte`

### PR DESCRIPTION
**Pull request by ribaraka**

### CODE REVIEW

File: internal/handlers/common/compare.go [hunk 1]
```diff
@@ -193,10 +193,31 @@ func compare(docValue, filter any) compareResult {
 
 	switch docValue := docValue.(type) {
 	case *types.Document:
-		// TODO: implement document comparing
+		if filter, ok := filter.(*types.Document); ok {
+			switch compareDocuments(filter, docValue) {
```
Comment (seeforschauer) [inline internal/handlers/common/compare.go:197]: here you can probably return `compareResults`, aren't you? 
and linter check will pass.

**Comment by AlekSi** [comment]
@ribaraka Let's put on pause for now. I'm currently investigating the way MongoDB compares and sorts values. It seems that we could drop `notEqual` completely and do some other simplifications. I will try to be back with some results in a few days.

**Review by AlekSi — COMMENTED** [review]
Please re-request review once ready

**Comment by w84thesun** [comment]
@ribaraka is this PR is still relevant?

**Comment by w84thesun** [comment]
@AlekSi should we close this one?

**Comment by AlekSi** [comment]
It is covered now.

Thank you for your contributions.

**Pull request closed without merging**